### PR TITLE
[BPK-1110] Add state management examples to horizontal nav readmes

### PR DESCRIPTION
**Pull request by shaundon**
This is to make it easier for consumers to use the documentation.

### CODE REVIEW

File: native/packages/react-native-bpk-component-horizontal-nav/readme.md [hunk 1]
```diff
@@ -12,28 +12,41 @@ npm install react-native-bpk-component-horizontal-nav --save-dev
...
-const styles = StyleSheet.create({
-  container: {
-    flex: 1,
-    justifyContent: 'center',
-    padding: spacingBase,
+export default class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      selectedId: 'flights'
```
Comment (matteo-hertel) [inline native/packages/react-native-bpk-component-horizontal-nav/readme.md]: there is a missing comma according to es-lint

File: native/packages/react-native-bpk-component-horizontal-nav/readme.md [hunk 2]
```diff
@@ -12,28 +12,41 @@ npm install react-native-bpk-component-horizontal-nav --save-dev
...
-          <BpkHorizontalNavItem id="nav-1" title="Item Two" onPress={() => {}} />
-          <BpkHorizontalNavItem id="nav-2" title="Item Three" onPress={() => {}} />
-        </BpkHorizontalNav>
-      </View >
+      <BpkHorizontalNav selectedId={this.state.selectedId}>
+        <BpkHorizontalNavItem
+          id="flights"
+          title="Flights"
+          onPress={() => {
+            this.setState({ selectedId: 'flights'})
```
Comment (matteo-hertel) [inline native/packages/react-native-bpk-component-horizontal-nav/readme.md]: there is a missing semicolon according to es-lint

File: native/packages/react-native-bpk-component-horizontal-nav/readme.md [hunk 3]
```diff
@@ -12,28 +12,41 @@ npm install react-native-bpk-component-horizontal-nav --save-dev
...
+          }}
+        />
+        <BpkHorizontalNavItem
+          id="hotels"
+          title="Hotels"
+          onPress={() => {
+            this.setState({ selectedId: 'hotels'})
```
Comment (matteo-hertel) [inline native/packages/react-native-bpk-component-horizontal-nav/readme.md]: here too

File: native/packages/react-native-bpk-component-horizontal-nav/readme.md [hunk 4]
```diff
@@ -12,28 +12,41 @@ npm install react-native-bpk-component-horizontal-nav --save-dev
...
+          }}
+        />
+        <BpkHorizontalNavItem
+          id="car-hire"
+          title="Car hire"
+          onPress={() => {
+            this.setState({ selectedId: 'car-hire'})
```
Comment (matteo-hertel) [inline native/packages/react-native-bpk-component-horizontal-nav/readme.md]: and the final one missing here

**Comment by shaundon** [comment]
@matteo-hertel thanks for the review! I've updated now.

**Pull request merged**